pkg/cookies/cookies.go

```diff
@@ -39,7 +40,29 @@ func MakeCookie(req *http.Request, name string, value string, path string, domai
 // MakeCookieFromOptions constructs a cookie based on the given *options.CookieOptions,
 // value and creation time
 func MakeCookieFromOptions(req *http.Request, name string, value string, opts *options.CookieOptions, expiration time.Duration, now time.Time) *http.Cookie {
-	return MakeCookie(req, name, value, opts.CookiePath, opts.CookieDomain, opts.CookieHTTPOnly, opts.CookieSecure, expiration, now, ParseSameSite(opts.CookieSameSite))
+	host := req.Header.Get("X-Forwarded-Host")
+	if host == "" {
+		host = req.Host
+	}
+
+	// Sort cookie domains by length, so that we try longer (and more specific)
+	// domains first
+	sortedDomains := opts.CookieDomains
+	sort.Slice(sortedDomains, func(i, j int) bool {
+		return len(sortedDomains[i]) > len(sortedDomains[j])
+	})
+	for _, domain := range sortedDomains {
```

Comment:
Can any of this logic be extracted, it looks very similar to where it's implemented elsewhere?